repo.sh

```diff
@@ -287,6 +309,239 @@ status ()
+    ["frontend-app-communications"]="edx"
+    ["frontend-app-discussions"]="edx"
+    ["frontend-app-profile"]="edx"
+    ["frontend-app-enterprise-public-catalog"]="edx"
+    ["frontend-app-publisher"]="edx"
+    ["frontend-app-support-tools"]="edx"
+    ["frontend-app-admin-portal"]="edx"
+    ["frontend-app-learner-portal-enterprise"]="edx"
+    ["frontend-app-enterprise-checkout"]="edx"
+    ["frontend-app-authoring"]="edx"
+    ["frontend-app-instruct"]="edx"
+    ["frontend-app-catalog"]="edx"
+    ["openedx-translations"]="edx"
+    ["frontend-app-payment"]="edx"
+    ["edx-analytics-data-api"]="edx"
+    ["enterprise-catalog"]="openedx"
+    ["portal-designer"]="edx"
+    ["license-manager"]="openedx"
+    ["codejail-service"]="openedx"
+    ["enterprise-access"]="openedx"
+    ["frontend-app-authn"]="openedx"
+    ["frontend-app-course-authoring"]="openedx"
+    ["registrar"]="edx"
+    ["frontend-app-account"]="openedx"
+    ["enterprise-subsidy"]="openedx"
+    ["edx-exams"]="edx"
+)
+
+setup_forked_repo_remotes ()
+{
+    local repo_name=$1
+    local expected_primary_org=${FORKED_REPOS[$repo_name]}
+    local edx_remote_exists
+    local openedx_remote_exists
+    local origin_exists
+    local origin_url=""
+    local origin_org=""
+    local other_org
+    local other_remote_exists
+    local other_url
+    
+    # Check if we're in a git repository
+    if [ ! -d ".git" ]; then
+        echo "ERROR: $repo_name is not a git repository"
+        return 1
+    fi
+    
+    # Check if both remotes already exist (idempotency check)
+    edx_remote_exists=$(git remote | grep "^edx$" || true)
+    openedx_remote_exists=$(git remote | grep "^openedx$" || true)
+    origin_exists=$(git remote | grep "^origin$" || true)
+    
+    if [ -n "$edx_remote_exists" ] && [ -n "$openedx_remote_exists" ] && [ -z "$origin_exists" ]; then
+        echo "Both edx and openedx remotes already exist in $repo_name. No changes needed."
+        return 0
+    fi
+    
+    echo "Setting up remotes for forked repository: $repo_name"
+    
+    # Try to get origin URL first
+    origin_url=$(git remote get-url origin 2>/dev/null || true)
+    
+    if [ -n "$origin_url" ]; then
+        # Origin exists, determine its organization
+        if [[ $origin_url =~ github\.com[:/]edx/ ]]; then
+            origin_org="edx"
+        elif [[ $origin_url =~ github\.com[:/]openedx/ ]]; then
+            origin_org="openedx"
+        else
+            echo "ERROR: Unexpected origin URL in $repo_name: $origin_url"
+            echo "Expected URL to be from either edx or openedx organization"
+            return 1
+        fi
+        
+        # Rename origin to the correct organization name if not already done
+        if [ -z "$(git remote | grep "^${origin_org}$")" ]; then
+            echo "Renaming origin to '$origin_org' in $repo_name"
+            git remote rename origin "$origin_org"
+            if [ $? -ne 0 ]; then
```

Comment:
Bits like this can be simplified to:

```suggestion
            if ! git remote rename origin "$origin_org"; then
```

Reply:
With running shellcheck , the suggested bits which can be simplified went onto simplify those. Thank you.